ui/src/app.js

```diff
@@ -650,14 +656,135 @@ function setWifiList (recs, requestedPage) {
  * @param {String} data.cancel
  */
 function setTermsScreen (data) {
-  const $screen = $('.js-terms-screen')
-
+  var $screen = $('.js-terms-screen')
   $screen.find('.js-terms-title').html(data.title)
-  $screen.find('.js-terms-text').html(data.text)
+  renderTermsText(data.text)
   $screen.find('.js-terms-accept-button').html(data.accept)
   $screen.find('.js-terms-cancel-button').html(data.cancel)
 }
 
+// to temove <p> & </p> tag from text
+function removePTagFromText (text) {
+  return text.slice(3, text.length - 4)
+}
+
+function splitToLines (textArray) {
+  var linesArray = []
+  var lineEndIndex = 0
+  while (lineEndIndex <= textArray.length - 2) {
+    var lineStartIndex = lineEndIndex + 1
+    if (lineEndIndex !== 0) {
+      lineEndIndex = calculateLinesLastWordIndex(textArray, lineEndIndex + 1)
+    } else {
+      lineStartIndex = 0
+      lineEndIndex = calculateLinesLastWordIndex(textArray, lineEndIndex)
+    }
+    linesArray.push(createLine(textArray, lineStartIndex, lineEndIndex))
+  }
+  return linesArray
+}
+
```

Comment:
This splitting is problematic since we don't know how the browser will format the text. Instead, I think we should use the ``scrollTo()`` JavaScript method to just scroll the text down and up one screen (minus a line or two). The div with the text can have a fixed height, and have ``overflow: hidden``.